diff --git a/TMessagesProj/src/main/java/org/telegram/ui/MainTabsActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/MainTabsActivity.java
--- a/TMessagesProj/src/main/java/org/telegram/ui/MainTabsActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/MainTabsActivity.java
@@ -242,29 +242,33 @@ public class MainTabsActivity extends ViewPagerActivity implements NotificationC
     private boolean tabletLayout;
     public void updateLayout() {
-//        if (tabletLayout == AndroidUtilities.isTablet()) return;
-//        tabletLayout = AndroidUtilities.isTablet();
-//
-//        final boolean isUpdateLayoutVisible = updateLayoutWrapper.isUpdateLayoutVisible();
-//        final int updateLayoutHeight = isUpdateLayoutVisible ? dp(UpdateLayoutWrapper.HEIGHT) : 0;
-//        int bottomMargin = isUpdateLayoutVisible ? (navigationBarHeight + updateLayoutHeight) : 0;
-//        if (tabletLayout) {
-//            bottomMargin = Math.max(bottomMargin, navigationBarHeight + dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS));
-//        }
-//        final FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(LayoutHelper.MATCH_PARENT, LayoutHelper.MATCH_PARENT, Gravity.FILL);
-//        if (tabletLayout) {
-//            lp.leftMargin = dp(6);
-//            lp.rightMargin = dp(6);
-//            lp.topMargin = dp(6);
-//        }
-//        lp.bottomMargin = bottomMargin;
-//
-//        viewPager.setLayoutParams(lp);
-//        viewPager.setTabletLayout(tabletLayout);
-//        checkUi_fadeView();
+        final boolean isTablet = AndroidUtilities.isTablet();
+        if (tabletLayout == isTablet && viewPager != null) return;
+        tabletLayout = isTablet;
+
+        final boolean isUpdateLayoutVisible = updateLayoutWrapper != null && updateLayoutWrapper.isUpdateLayoutVisible();
+        final int updateLayoutHeight = isUpdateLayoutVisible ? dp(UpdateLayoutWrapper.HEIGHT) : 0;
+        int bottomMargin = isUpdateLayoutVisible ? (navigationBarHeight + updateLayoutHeight) : 0;
+        if (tabletLayout) {
+            bottomMargin = Math.max(bottomMargin, navigationBarHeight + dp(DialogsActivity.MAIN_TABS_HEIGHT_WITH_MARGINS));
+        }
+        final FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(LayoutHelper.MATCH_PARENT, LayoutHelper.MATCH_PARENT, Gravity.FILL);
+        if (tabletLayout) {
+            lp.leftMargin = dp(6);
+            lp.rightMargin = dp(6);
+            lp.topMargin = dp(6);
+        }
+        lp.bottomMargin = bottomMargin;
+
+        if (viewPager != null) {
+            viewPager.setLayoutParams(lp);
+            viewPager.setTabletLayout(tabletLayout);
+        }
+        checkUi_fadeView();
     }
